platforms/minestom/src/main/java/com/dfsek/terra/minestom/block/MinestomBlockState.java

```diff
@@ -0,0 +1,104 @@
+package com.dfsek.terra.minestom.block;
+
+import com.dfsek.terra.api.block.BlockType;
+import com.dfsek.terra.api.block.state.BlockState;
+import com.dfsek.terra.api.block.state.properties.Property;
+
+import net.minestom.server.instance.block.Block;
+
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+
+public class MinestomBlockState implements BlockState {
+    private final Block block;
+
+    public MinestomBlockState(Block block) {
+        if (block == null) {
+            this.block = Block.AIR;
+        } else {
+            this.block = block;
+        }
+    }
+    
+    public MinestomBlockState(String data) {
+        if (!data.contains("[")) {
+            block = Block.fromNamespaceId(data);
+            return;
+        }
+        
+        String[] split = data.split("\\[");
+        String namespaceId = split[0];
+        String properties = split[1].substring(0, split[1].length() - 1);
+        Block block = Block.fromNamespaceId(namespaceId);
+        HashMap<String, String> propertiesMap = new HashMap<>();
+
+        for (String property : properties.split(",")) {
+            String[] kv = property.split("=");
+            propertiesMap.put(kv[0].strip(), kv[1].strip());
+        }
+
+        assert block != null;
+        this.block = block.withProperties(propertiesMap);
+    }
```

Comment:
Is this constructor ever used?
If not, then it should be removed.

Reply:
Used by `MinestomWorldHandle#createBlockState`